### PR TITLE
Comply with guide for method doc: hash.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**
This is to move some RDoc into further compliance with https://github.com/ruby/ruby/blob/master/doc/method_documentation.rdoc.  I view this a mergeable if compliance is improved, and if there are no new errors.

Methods:
- ::new
- ::[]
- ::try_convert
- #rehash
- #[]
- #fetch
- #default
- #default=
- #default_proc
- #default_proc=

### CODE REVIEW

File: hash.c [hunk 1]
```diff
@@ -1938,28 +1884,14 @@ rb_check_hash_type(VALUE hash)
...
+ *    Hash.try_convert(obj) -> obj, new_hash, or nil
  *
- *  Returns <tt>nil</tt> unless <tt>obj.respond_to?(:to_hash)</tt>.
+ *  If +obj+ is a \Hash object, returns +obj+.
  *
- *  ---
+ *  Otherwise if +obj+ responds to <tt>:to_hash</tt>,
+ *  calls <tt>obj.to_hash</tt> and returns the result.
  *
- *  Raises an exception unless <tt>obj.to_hash</tt> returns a \Hash object:
```
Comment (jeremyevans) [inline hash.c:1955]: I would keep this line but move it to the bottom (still drop the example).

File: hash.c [hunk 2]
```diff
@@ -2171,29 +2094,25 @@ rb_hash_lookup(VALUE hash, VALUE key)
 
 /*
  *  call-seq:
- *    hash.fetch(key) -> value
- *    hash.fetch(key, default) -> value
+ *    hash.fetch(key, default_value = self.default) -> value
```
Comment (jeremyevans) [inline hash.c:2097]: I think this is incorrect.  For one, the description below states that the method does not use the values of method `default`, so it couldn't be the default value of the 2nd argument.  I would keep these three call-seq lines separate.

File: hash.c [hunk 3]
```diff
@@ -2240,23 +2159,21 @@ rb_hash_fetch(VALUE hash, VALUE key)
 
 /*
  *  call-seq:
- *    hash.default -> value
- *    hash.default(key) -> value
+ *    hash.default(key = self.default) -> value
```
Comment (jeremyevans) [inline hash.c:2162]: I don't think these two lines can be combined.  The behavior in the 0 argument case when a default proc is used is to always return nil.  The behavior in the 1 argument case when a default proc is used is to call the default proc with the key.  The behavior of the 0 argument case with the default proc should be documented (no need for an example).

File: hash.c (hunk 1 above)
Comment (BurdetteLamar) [reply hash.c:1955]: Fixed.

File: hash.c (hunk 2 above)
Comment (BurdetteLamar) [reply hash.c]: Fixed.

File: hash.c (hunk 3 above)
Comment (BurdetteLamar) [reply hash.c]: Fixed.

**Pull request merged**